ml-agents/mlagents/trainers/trajectory.py

```diff
@@ -0,0 +1,102 @@
+from typing import List, NamedTuple
+import numpy as np
+
+from mlagents.trainers.buffer import AgentBuffer
+
+
+class AgentExperience(NamedTuple):
+    obs: List[np.ndarray]
+    reward: float
+    done: bool
+    action: np.ndarray
+    action_probs: np.ndarray
+    action_pre: np.ndarray  # TODO: Remove this
+    action_mask: np.ndarray
+    prev_action: np.ndarray
+    max_step: bool
+    memory: np.ndarray
+
+
+class SplitObservations(NamedTuple):
+    vector_observations: np.ndarray
+    visual_observations: List[np.ndarray]
+
+
+class Trajectory(NamedTuple):
+    steps: List[AgentExperience]
+    next_obs: List[
+        np.ndarray
+    ]  # Observation following the trajectory, for bootstrapping
+    agent_id: str
+
+
+def split_obs(obs: List[np.ndarray]) -> SplitObservations:
```

Comment:
Maybe make this a static method of SplitObservations, e.g. SplitObservations.from_observations()

Reply:
So somewhat related but I've run into this problem before - in Python 3.6 if you put the method as part of the class, the type annotation (of that same class) no longer works. 

One solution I've found is to use a string (e.g. `"SplitObservations"` as the type annotation. Is this correct?

Reply:
Yup, that's the recommendation https://mypy.readthedocs.io/en/latest/cheat_sheet_py3.html#miscellaneous

Reply:
Done